### PR TITLE
✨ Add e2e test and template for K8s version upgrade

### PR DESCRIPTION
**Pull request by Jont828**
**What this PR does / why we need it**: Add e2e test and template for K8s version upgrade to verify that version upgrade works.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

**Comment by Jont828** [comment]
@willie-yao @mboersma Could you guys take a look at this?

File: test/e2e/data/addons-helm/v1beta1/bases/cluster-with-topology.yaml [hunk 1]
```diff
@@ -0,0 +1,78 @@
+apiVersion: cluster.x-k8s.io/v1beta1
```
Comment (mboersma) [inline test/e2e/data/addons-helm/v1beta1/bases/cluster-with-topology.yaml:1]: Is this file `cluster-with-topology.yaml` referenced anywhere?

File: test/e2e/data/addons-helm/v1beta1/cluster-template.yaml [hunk 2]
```diff
@@ -10,7 +10,7 @@ spec:
   namespace: tigera-operator
   releaseName: projectcalico
   repoURL: https://docs.tigera.io/calico/charts
-  valuesTemplate: |-
+  valuesTemplate: |
```
[the remark this answers is not in the document]
Comment (Jont828) [reply test/e2e/data/addons-helm/v1beta1/cluster-template.yaml:13]: The `-` removes the newlines at the end of the block. Shouldn't make a difference but it'll be consistent with the other templates

File: test/e2e/data/addons-helm/v1beta1/bases/cluster-with-topology.yaml (hunk 1 above)
Comment (Jont828) [reply test/e2e/data/addons-helm/v1beta1/bases/cluster-with-topology.yaml:1]: Yes, it's used in kustomzation.yaml for the upgrade template.

**Comment by Jont828** [comment]
@willie-yao The process right now would be to pull the Cluster and ClusterClass template from CAPI and manually change the pod security and the label.

**Review by willie-yao — APPROVED** [review]
/lgtm

**Review by mboersma — APPROVED** [review]
/lgtm++

**Pull request merged**